lib/battle/BattleInfo.h

```diff
@@ -131,13 +136,16 @@ class DLL_LINKAGE BattleInfo : public CBonusSystemNode, public CBattleInfoCallba
 	const CGHeroInstance * getHero(PlayerColor player) const; //returns fighting hero that belongs to given player
 
 	void localInit();
-
+	static void adjustOppositeBonuses(BattleInfo * curB);
 	static BattleInfo * setupBattle(int3 tile, ETerrainType terrain, BFieldType battlefieldType, const CArmedInstance * armies[2], const CGHeroInstance * heroes[2], bool creatureBank, const CGTownInstance * town);
 
 	ui8 whatSide(PlayerColor player) const;
 
 	static BattlefieldBI::BattlefieldBI battlefieldTypeToBI(BFieldType bfieldType); //converts above to ERM BI format
 	static int battlefieldTypeToTerrain(int bfieldType); //converts above to ERM BI format
+
+private:
+	STRONG_INLINE static void adjustOppositeBonuses(CBonusSystemNode * node, PlayerColor ownerColor);

```

Comment:
Is this STRONG_INLINE modifier necessary?

Reply:
It's proposed to be called in the for() many times

Reply:
It's plain wrong - strong inline should only be used if you know very well what you are doing and only if detailed profiling tells you to do so. Using it mean you know better than compiler how to optimize this case. 99% of the time you will want to just use "inline" or no modifier at all

Reply:
What are particular objections? All that code moved to the separate function only for readability and I don't want to waste resources for  some kind of  push / call / pop / ret assembler instructions.

Reply:
objection is about using STRONG_INLINE - skip it unless you know what you're doing and did performance profiling compared to build with compiler optimization on

Reply:
In the case of principal approach, release build needs to be checked for all supported platforms, there are no resources for this